src/metabase/driver/sql/query_processor.clj

```diff
@@ -8,67 +8,38 @@
             [metabase.query-processor.error-type :as qp.error-type]
-            [metabase.query-processor.interface :as i]
             [metabase.query-processor.middleware.annotate :as annotate]
             [metabase.query-processor.middleware.wrap-value-literals :as value-literal]
             [metabase.query-processor.store :as qp.store]
+            [metabase.query-processor.util.add-alias-info :as add]
+            [metabase.query-processor.util.nest-query :as nest-query]
             [metabase.util :as u]
             [metabase.util.honeysql-extensions :as hx]
             [metabase.util.i18n :refer [deferred-tru tru]]
-            [metabase.util.schema :as su]
-            [potemkin.types :as p.types]
+            [potemkin :as p]
             [pretty.core :refer [PrettyPrintable]]
             [schema.core :as s])
   (:import metabase.models.field.FieldInstance
            [metabase.util.honeysql_extensions Identifier TypedHoneySQLForm]))
 
-;; TODO - yet another `*query*` dynamic var. We should really consolidate them all so we only need a single one.
-(def ^:dynamic ^:private *query*
-  "The INNER query currently being processed, for situations where we need to refer back to it."
-  nil)
-
-(def ^:dynamic ^:private *source-aliases*
-  "A map of field-clause -> alias for Fields that come from nested source queries or joins, e.g.
-
-    {[:field-id 1 {:join-alias \"X\"}] \"Q1__my_field\"}
-
-  This is needed because source queries and joins will generate their own 'unambiguous alias' for Fields that come
-  from their own source queries or joins, and we need to use that alias when referring to the Field in the parent
-  query. e.g.
-
-    -- all uses of `*category_id` may well be the same column, but we need to use the correct alias based on the one
-    -- used in nested queries or joins
-    SELECT J.t3__category_id AS J__category_id
-    FROM T1
-    LEFT JOIN (
-      SELECT T2.category_id AS category_id,
-             T3.category_id AS T3__category_id
-      FROM T2
-      LEFT JOIN T3
-      ON T2.some_id = t3.id
-    ) J
-    ON T1.id = J1.T3__category_id
-
-  See #16254 for more information."
-  nil)
+(def source-query-alias
+  "Alias to use for source queries, e.g.:
 
-(def ^:dynamic *nested-query-level*
-  "How many levels deep are we into nested queries? (0 = top level.) We keep track of this so we know what level to
-  find referenced aggregations (otherwise something like [:aggregation 0] could be ambiguous in a nested query).
-  Each nested query increments this counter by 1."
-  0)
+    SELECT source.*
+    FROM ( SELECT * FROM some_table ) source"
+  "source")
 
-(def ^:dynamic *field-options*
-  "Bound to the `options` part of a `:field` clause when that clause is being compiled to HoneySQL. Useful if you store
-  additional keys there and need to access them."
+(def ^:dynamic *query*
+  "The INNER query currently being processed, for situations where we need to refer back to it."
```

Comment:
Does it make sense to call this `*inner-query*` then?

Reply:
Yeah, it probably does. I'll change it

Reply:
fixed